mParticle_AppboyTests/mParticle_AppboyTests.m

```diff
@@ -82,6 +82,43 @@ - (void)testStartwithAdvancedConfig {
     XCTAssertEqualObjects(optionsDictionary, testOptionsDictionary);
 }
 
+- (void)testEndpointOverride {
+    MPKitAppboy *appBoy = [[MPKitAppboy alloc] init];
+    
+    NSDictionary *kitConfiguration = @{@"apiKey":@"BrazeID",
+                                       @"host":@"https://test\\.org",
```

Comment:
Why the two slashes here? The happy-path/possible cases that configuration will currently pass down for `host` are:

- `sdk.iad-01.braze.com`
- `sdk.iad-02.braze.com`
- `sdk.iad-03.braze.com`
- `sdk.fra-01.braze.eu`

In addition to testing different host values, can you update this test to be a little more robust for what's passed into `getApiEndpoint`? For example testing some variations that could get passed such as:

- nil
- https://foo.bar.com
- http://foo.bar.com
- foo.bar.com
- https://foo.bar.com/baz
- https://foo.bar.com/baz/baz

Reply:
The slashes after 'test' got leftover from regex styling (to mean . you have to both '\' escape '\' and '.'

You can see that here '@"https.*\\.com"' and also see why that didn't cause a problem for the regex. '.*' means any character any number of times.

I did add one for nil since using a nil host as template will cause the regex method to crash. But I'll add others for robustness sake.

Reply:
i mean what happens if nil (and all of the above) get passed in to getApiEndpoint as the argument, not the host.

Reply:
@samdozor Just saw that you also meant passing into `getApiEndpoint` The nil will crash since this is a nonnull method.

These will result in no changes to the string that being passed in because they don't fit the regex pattern.

- foo.bar.com
- http://foo.bar.com